docs/adr/adr-014-bootstrap-from-previous-peers.md

```diff
@@ -0,0 +1,557 @@
+
+</br>
+
+## Approach B: Periodic sampling of peers from the peerstore for "good peers" selection using liveness buckets
+
+In this approach, we will periodically sample peers from the host's `Peerstore` and select the "good peers", with the "good peers" being the ones that have been connected to the longest possible period of time.
+
+Taking inspiration from [this proposition](https://github.com/libp2p/go-libp2p-kad-dht/issues/254#issuecomment-633806123) we suggest to look for peers that have been continuously online, such that, for the given buckets of connection durations:
+
+* < 1 hour ago
+* < 1 day ago
+* < 1 week ago
+* < 1 month ago.
+
+the "good enough peers" to select should be present in most recent buckets, and should be peers we've seen often. For example, if a peer has been connected to the node for 1 day, then it should figure up in the buckets: "< 1 hour" and , "< 1 day", hence making it a "good enough peer" to select for persistence.
+
+The bucketing logic that triages peers into the mentioned buckets is executed periodically, with a new list of "good enough peers" being selected and persisted at that same time, and also at the moment of node shutdown.
+
+## Implementation
+
+### Storage
+
+We suggest to use the same storage mechanism proposed in approach A.
+
+### Peer Selection
+
+To achieve bucket-powered peer selection, we propose to create a standalone cache like component to manage buckets, such that it tracks peers by their IDs (_`peer.ID`_) and puts them in the correct bucket every time a `Sort` method is called on the component. The component, thus, maps each peer ID to a list of timestamps, each timestamp marking when did the bucketing component see this peer.
+
+The length of the list of timestamps will be equal to the amount of times the peer was seen, and the dates from oldest to newest will help the component sort the peer into appropriate buckets.
+
+Example:
+A list of 1 timestamp means it's been seen once, and if the timestamp was 3 hours ago, then the peer would be sorted into the "< 1 day ago" bucket.
+
+Depending on how many peers the node will be connected to, this could be the best peer it can get, or the worst.
+
+A better example of what a very good peer would be is:
+A peer that has a list of timestamps whose length is equal to 10, and the oldest timestamp was a week ago and the newest from an hour ago. This will make the peer fall into "< 1 week ago" bucket but also all the other recent buckets.
+
+We suggest to also turn the "amount of times seen" into buckets, such that we sort peers into buckets of "how many times" they were seen, modifying the selection criteria for good peers to become:
+
+* Peers that fall in most recent buckets + Peers that fall in highest seen count buckets
+
+**The interface**: We will go with the name `Bucketer` for the lack of a better one for the moment
+
+```go
+type Bucketer interface {
```

Comment:
Why does this need an interface?

Reply:
That's the best way I could use to describe a behavior in this ADR